ffi/build.rs

```diff
@@ -1,12 +1,220 @@
+    }
+}
+
+/// in order to re-run this (i.e. for new module compilation), you could:
+/// 1. cargo clean
+/// 2. e.g. cargo build/run --features nb2/build-kni
+fn build_dpdk(rte_sdk: &str) {
+    let cflags =
+        env::var("CFLAGS").unwrap_or_default() + " -g3 -Wno-error=maybe-uninitialized -fPIC";
+    let config = env::var("DPDK_CONFIG").unwrap_or("x86_64-native-linuxapp-gcc".to_string());
+
+    #[cfg(feature = "build-kni")]
+    set_dpdk_kni(&config);
+
+    #[cfg(feature = "build-pcap")]
+    set_dpdk_pcap(&config);
+
+    Command::new("make")
+        .current_dir(Path::new("../dpdk"))
+        .arg("config")
+        .arg(format!("T={}", config))
+        .arg(&format!("EXTRA_CFLAGS={}", cflags))
+        .status()
+        .expect("Couldn't configure dpdk for compilation!");
+
+    Command::new("make")
+        .current_dir(Path::new("../dpdk"))
+        .args(&["-j", &format!("{}", nproc())])
+        .arg("install")
+        .arg(format!("T={}", config))
+        .arg(&format!("DESTDIR={}/build", rte_sdk))
+        .arg(&format!("EXTRA_CFLAGS={}", cflags))
+        .status()
+        .expect("Couldn't install dpdk!");
+}
+
+fn setup_dpdk() {
+    let driver = env::var("DPDK_DRIVER").unwrap_or("uio_pci_generic".to_string());
+    let devices_env = env::var("DPDK_DEVICES");
+
+    // Note: no `lsmod` check here as uio_pci_generic is auto-loaded on some
+    // systems, but not yet bound.
+    if let Ok(devices) = devices_env {
+        Command::new("python")
+            .arg("../dpdk/usertools/dpdk-devbind.py")
+            .arg("--force")
+            .arg("-b")
+            .arg(driver)
+            .args(devices.split_whitespace())
+            .status()
+            .expect("Couldn't devbind driver to devices!");
+    } else {
+        println!(
+            "Please make sure to run `dpdk-devbind` and bind devices to a DPDK-compatible driver"
+        );
+    }
+}
+
+fn make_dpdk(rte_sdk: &str) {
+    build_dpdk(&rte_sdk);
+    setup_dpdk();
+
+    #[cfg(feature = "build-kni")]
+    setup_kni(&rte_sdk);
+}
+
+#[cfg(feature = "build-kni")]
+fn setup_kni(rte_sdk: &str) {
+    if !is_kni_bound() {
+        let insmod = Command::new("sudo")
+            .arg("insmod")
+            .arg(format!(
+                "{}/build/lib/modules/{}/extra/dpdk/rte_kni.ko",
+                rte_sdk,
+                to_str(
+                    &uname()
+                        .expect("Can't find kernel version via `uname`")
+                        .release
+                )
+            ))
+            .output()
+            .expect("Could not retrieve `insmod`'s output!")
+            .status
+            .success();
+
+        if !insmod {
+            panic!("Kmod module not inserted")
+        }
+    }
+}
+
+fn bind(path: &Path, rte_sdk: &str) {
     cc::Build::new()
         .file("src/shim.c")
-        .include("/opt/dpdk/build/include")
+        .include(format!("{}/build/include", rte_sdk))
```

Comment:
done to keep step w/ utils repo right now, but we can take this in steps for a better change there too, for the container. /cc @drunkirishcoder